In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Laboratorio 1 - Exploración, preparación y regresión lineal

## 1. Comprensión y carga de los datos

En esta primera etapa se cargan los conjuntos de entrenamiento y prueba, junto con el diccionario de datos. El objetivo es comprender la estructura de los archivos antes de realizar cualquier transformación o corrección.

In [3]:
datos = pd.read_csv("Datos Lab 1.csv")
datos_test = pd.read_csv("Datos Test Lab 1.csv")
diccionario = pd.read_excel("Diccionario de datos.xlsx")

In [4]:
print("Dimensiones entrenamiento:", datos.shape)
print("Dimensiones test:", datos_test.shape)
print("Dimensiones diccionario:", diccionario.shape)

Dimensiones entrenamiento: (2576, 27)
Dimensiones test: (364, 26)
Dimensiones diccionario: (27, 3)


In [5]:
datos.head()

,fecha,presion_media,presion_min,presion_max,presion_desv,humedad_media,humedad_min,humedad_max,humedad_desv,viento_media,...,viento_norte,viento_este,direccion_viento,registros_del_dia,anio,dia_del_anio,estacion_anio,mes,sector_viento,temp_max_manana
0,2009-01-01,999.1456,996.50,1000.87,1.3993,0.910860,0.875000,94.8,1.7650,0.7786,...,-0.3618,-0.0223,183.5308,143.0,2009.0,1.0,invierno,January,S,-2.12
1,2009-01-02,999.6006,997.93,1002.65,1.5039,0.920868,86.600000,96.3,2.7588,1.4195,...,0.4267,0.3689,40.8436,144.0,2009.0,2.0,invierno,JULY,NE,-0.82
2,2009-01-03,998.5486,993.05,1002.49,3.1304,76.458100,48.390000,93.9,15.1796,1.2509,...,-0.6993,-0.5268,216.9916,144.0,2009.0,3.0,invierno,January,SO,-0.63
3,2009-01-04,988.5107,985.12,992.93,2.3223,89.417400,97.946704,NaN,4.4904,1.7204,...,-1.1268,-1.0413,222.7419,144.0,2009.0,4.0,invierno,JANUARY,SO,-1.44
4,2009-01-05,990.4057,NaN,997.54,4.2315,86.260400,74.600000,93.2,5.3922,3.8003,...,2.6275,0.2874,6.2418,NaN,2009.0,5.0,East,January,N,-10.88


In [6]:
datos_test.head()

,fecha,presion_media,presion_min,presion_max,presion_desv,humedad_media,humedad_min,humedad_max,humedad_desv,viento_media,...,rafaga_desv,viento_norte,viento_este,direccion_viento,registros_del_dia,anio,dia_del_anio,estacion_anio,mes,sector_viento
0,01.01.2016,998.9860,995.57,1000.60,1.1082,96.1493,90.8,98.7,2.0034,0.9649,...,0.8415,-0.8328,0.0927,173.6501,144,2016,1,invierno,January,S
1,02.01.2016,991.4241,989.27,995.51,1.9413,92.3236,85.8,96.8,2.6562,1.9886,...,1.3604,1.1161,0.6688,30.9301,144,2016,2,invierno,January,NE
2,03.01.2016,984.0232,974.12,989.39,4.5841,83.6056,73.9,94.7,6.9967,2.4992,...,1.2558,-0.7923,1.2638,122.0847,144,2016,3,invierno,January,SE
3,04.01.2016,968.2449,966.52,974.02,1.9034,85.4688,75.5,93.8,5.8148,2.0932,...,1.4544,-1.6217,0.7020,156.5944,144,2016,4,invierno,January,SE
4,05.01.2016,970.0099,967.85,972.90,1.6668,94.3521,92.0,96.8,1.4253,1.8380,...,0.7284,1.6549,0.7109,23.2458,144,2016,5,invierno,January,NE


In [7]:
diccionario

,variable,tipo,descripcion
0,fecha,texto,"Fecha de la observación, en formato día.mes.año."
1,presion_media,numérico,Presión atmosférica media del día (mbar).
2,presion_min,numérico,Presión atmosférica mínima del día (mbar).
3,presion_max,numérico,Presión atmosférica máxima del día (mbar).
4,presion_desv,numérico,Desviación típica de la presión durante el día...
5,humedad_media,numérico,Humedad relativa media del día (%).
6,humedad_min,numérico,Humedad relativa mínima del día (%).
7,humedad_max,numérico,Humedad relativa máxima del día (%).
8,humedad_desv,numérico,Desviación típica de la humedad relativa (%).
9,viento_media,numérico,Velocidad media del viento durante el día (m/s).


In [8]:
columnas_entrenamiento = set(datos.columns)- set(datos_test.columns)
columnas_entrenamiento

{'temp_max_manana'}

### Variable objetivo

La variable objetivo del problema es `temp_max_manana`, correspondiente a la temperatura máxima del día siguiente en grados Celsius.

Por lo tanto, se trata de un problema de aprendizaje supervisado de regresión, ya que se dispone de una variable objetivo numérica continua en el conjunto de entrenamiento.

Eso está alineado con la definición del curso: una tarea de regresión busca aprender una función h: R^d -> R para predecir una respuesta continua.

In [12]:
datos.info()

<class 'pandas.DataFrame'>
RangeIndex: 2576 entries, 0 to 2575
Data columns (total 27 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   fecha              2504 non-null   str    
 1   presion_media      2501 non-null   float64
 2   presion_min        2502 non-null   float64
 3   presion_max        2512 non-null   float64
 4   presion_desv       2496 non-null   float64
 5   humedad_media      2502 non-null   float64
 6   humedad_min        2496 non-null   float64
 7   humedad_max        2490 non-null   float64
 8   humedad_desv       2515 non-null   float64
 9   viento_media       2490 non-null   float64
 10  viento_min         2485 non-null   float64
 11  viento_max         2495 non-null   float64
 12  viento_desv        2504 non-null   float64
 13  rafaga_media       2498 non-null   float64
 14  rafaga_min         2504 non-null   float64
 15  rafaga_max         2503 non-null   float64
 16  rafaga_desv        2497 non-null   

In [13]:
datos_test.info()

<class 'pandas.DataFrame'>
RangeIndex: 364 entries, 0 to 363
Data columns (total 26 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   fecha              364 non-null    str    
 1   presion_media      364 non-null    float64
 2   presion_min        364 non-null    float64
 3   presion_max        364 non-null    float64
 4   presion_desv       364 non-null    float64
 5   humedad_media      364 non-null    float64
 6   humedad_min        364 non-null    float64
 7   humedad_max        364 non-null    float64
 8   humedad_desv       364 non-null    float64
 9   viento_media       364 non-null    float64
 10  viento_min         364 non-null    float64
 11  viento_max         364 non-null    float64
 12  viento_desv        364 non-null    float64
 13  rafaga_media       364 non-null    float64
 14  rafaga_min         364 non-null    float64
 15  rafaga_max         364 non-null    float64
 16  rafaga_desv        364 non-null    fl

### Descripción inicial de los conjuntos

El conjunto de entrenamiento contiene 2.576 observaciones y 27 variables, mientras que el conjunto de prueba contiene 364 observaciones y 26 variables.

La diferencia corresponde a `temp_max_manana`, que únicamente está disponible en el conjunto de entrenamiento y constituye la variable objetivo del problema.

Cada observación representa información meteorológica agregada de un día. Las características incluyen medidas de presión atmosférica, humedad, velocidad y dirección del viento, ráfagas e información temporal.

Debido a que `temp_max_manana` es una variable numérica continua, el problema corresponde a una tarea de regresión dentro del aprendizaje supervisado.